.github/doctum/apiclient-services/doctum.js.twig

```diff
@@ -0,0 +1,353 @@
+                    }
+                });
+                Doctum.doctumSearchAutoComplete.addEventListener('results', function (event) {
+                    Doctum.markProgressFinished();
+                });
+            });
+        }
+        // Check if the lib is loaded
+        if (typeof autoComplete === 'function') {
+            Doctum.bootAutoComplete();
+        }
+    },
+    markInProgress: function () {
+            Doctum.doctumSearchAutoCompleteProgressBarContainer.className = 'search-bar';
+            Doctum.doctumSearchAutoCompleteProgressBar.className = 'progress-bar indeterminate';
+            if (typeof DoctumSearch === 'object' && DoctumSearch.pageFullyLoaded) {
+                DoctumSearch.doctumSearchPageAutoCompleteProgressBarContainer.className = 'search-bar';
+                DoctumSearch.doctumSearchPageAutoCompleteProgressBar.className = 'progress-bar indeterminate';
+            }
+    },
+    markProgressFinished: function () {
+        Doctum.doctumSearchAutoCompleteProgressBarContainer.className = 'search-bar hidden';
+        Doctum.doctumSearchAutoCompleteProgressBar.className = 'progress-bar';
+        if (typeof DoctumSearch === 'object' && DoctumSearch.pageFullyLoaded) {
+            DoctumSearch.doctumSearchPageAutoCompleteProgressBarContainer.className = 'search-bar hidden';
+            DoctumSearch.doctumSearchPageAutoCompleteProgressBar.className = 'progress-bar';
+        }
+    },
+    makeProgess: function () {
+        Doctum.makeProgressOnProgressBar(
+            Doctum.doctumSearchAutoCompleteProgressBarPercent,
+            Doctum.doctumSearchAutoCompleteProgressBar
+        );
+        if (typeof DoctumSearch === 'object' && DoctumSearch.pageFullyLoaded) {
+            Doctum.makeProgressOnProgressBar(
+                Doctum.doctumSearchAutoCompleteProgressBarPercent,
+                DoctumSearch.doctumSearchPageAutoCompleteProgressBar
+            );
+        }
+    },
+    loadAutoCompleteData: function (query) {
+        return new Promise(function (resolve, reject) {
+            if (Doctum.autoCompleteData !== null) {
+                resolve(Doctum.autoCompleteData);
+                return;
+            }
+            Doctum.markInProgress();
+            function reqListener() {
+                Doctum.autoCompleteLoading = false;
+                Doctum.autoCompleteData = JSON.parse(this.responseText).items;
+                Doctum.markProgressFinished();
+
+                setTimeout(function () {
+                    resolve(Doctum.autoCompleteData);
+                }, 50);// Let the UI render once before sending the results for processing. This gives time to the progress bar to hide
+            }
+            function reqError(err) {
+                Doctum.autoCompleteLoading = false;
+                Doctum.autoCompleteData = null;
+                console.error(err);
+                reject(err);
+            }
+
+            var oReq = new XMLHttpRequest();
```

Comment:
It is more common to use `fetch` now:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Reply:
The first comment explains this